[Ong Hee Jet] iP

heejet opened the pull request.

heejet committed:
Add Increment Level 0

heejet committed:
Renamed chatbot to Luke, add logos

heejet committed:
Add Level-1: Greet, Echo, Exit

heejet committed:
Add Level-2: Add, List

heejet committed:
Add Level 3: Mark as Done

heejet committed:
Change code to comply with SE-EDU Java coding standard

heejet committed:
Add Level-4: ToDo, Event, Deadline

ToDo is a sub class of Task that represents activities that do not
require a deadline.
Deadline is a sub class of Task that represents an activity that needs
to be completed by a specified deadline.
Event is a sub class of Deadline that represents and activity that
starts at a specific date/time and ends at a specific date/time.

heejet committed:
Add Level-4: ToDo, Event, Deadline

ToDo is a sub class of Task that represents activities that do not require a deadline.
Deadline is a sub class of Task that represents an activity that needs to be completed by a specified deadline.
Event is a sub class of Deadline that represents and activity that starts at a specific date/time and ends at a specific date/time.

heejet committed:
Add A-TextUiTesting

Semi-automated testing of Luke. Test input is stored in input.txt
and the expected output is stored in EXPECTED.TXT. Outputs from
Luke is redirected to ACTUAL.TXT and the script is located in
runtest.bat

heejet committed:
Add A-CodeQuality

Most of the code in the main method have more than 3 levels of
nesting. Multiple if-else statements causes the happy path to not
be prominent. Some varaible names like taskID arecommon across
classes but in reality they refer to different things.

The common behavior across code used for error handling result in
code duplication. Duplicate variables who do not refer to the same
object will lead to confusion.

Let's extract error handling code into their own method such as
handleOutOfBounds and handleInvalidCommand. Code to read in user
input and execute commands have also been extracted into their own
methods. Variable names should also be renamed to avoid confusion.

Extracting common behavior into a method increases the resuability
of code and makes the code easier to read. Variables have been
renamed to reduce confusion. For example, taskID in Luke.java have
been renamed to serialNumber to avoid being confused with taskID in
Task.java

heejet committed:
Add new test cases to input.txt and update EXPECTED.TXT

darrenlsx commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,237 @@
+            handleInvalidCommand();
+        }
+    }
+
```
I like how all the naming of the functions were easily understandable.

darrenlsx commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,237 @@
+            executeUnmarkTask(description);
+            break;
+        }
+    }
```
I like how you used a switch to make it clear and readable.

darrenlsx commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,237 @@
+        initialize();
+        readUserInput();
+        endProgram();
+    }
```
Neat and clean use of SLAP

darrenlsx commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,237 @@
+    private static ArrayList<String> commands;
+
+    /** Initializes all the objects used in LUKE, says "Hi" to the user */
+    private static void initialize() {
```
Perhaps in the function name can add a noun after initialize to know what are we initializing

SSzeWen commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,237 @@
+            }
+            else {
+                echoString(userInput);
+            }
```
Should the else statement in line 226 be on line 225 instead?

SSzeWen commented on src/main/java/TaskOrganizer.java:
```diff
@@ -0,0 +1,132 @@
+     * @param taskDate A string containing the dates of the task (E.g deadline or start and end date).
+     * @return True as task have been added without any errors, False if there are errors when adding the task.
+     */
+    public boolean addTask(String taskType, String taskName, String taskDate) {
```
Perhaps there could be a better naming here for addTask to sound like a boolean function

SSzeWen commented on src/main/java/Luke.java:
```diff
@@ -0,0 +1,237 @@
+        default:
+            executeUnmarkTask(description);
+            break;
+        }
```
This switch statement looks really clean and follows the Java Coding Standard well with no indentation for case clauses

SSzeWen commented on src/main/java/LukeLogo.java:
```diff
@@ -0,0 +1,58 @@
+     * Returns a randomized string corresponding to the name of the chatbot.
+     *
+     * @return String corresponding to an ASCII art of the string "LUKE".
+     */
```
Informative Comments that follows JavaDoc format

SSzeWen commented on src/main/java/TaskOrganizer.java:
```diff
@@ -0,0 +1,132 @@
+            newTask = new Event(taskName, newTaskID, startDate, endDate);
+            break;
+        default:
+            newTask = new Task(taskName, newTaskID);
```
I am not sure if there is a need to add a break statement here, or write //Fallthrough, since its the last statement

SSzeWen reviewed (commented):
Overall, the code is very neat and understandable that follows Java coding standards with little changes required. Comments are very nice and all the different classes are named well

heejet committed:
Add Level-5: Handle Errors

heejet committed:
Refactor code

heejet committed:
Merge commit 'ebdae8ac5b01e6e8fe2ac8a5876cc6459d1b2e6f'

heejet committed:
Merge branch 'branch-Level-5'

heejet committed:
Add A-Packages

Organize classes into suitable java packages.

heejet committed:
Add A-Packages

Organize classes into suitable java packages

heejet committed:
Merge branch 'branch-A-Packages'

heejet committed:
Add Level-6: Delete

heejet committed:
Add Level-7: Save

heejet committed:
Merge branch 'branch-Level-6'

heejet committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/luke/command/TaskOrganizer.java

heejet committed:
Bug fixes and refactor code

heejet committed:
Update runtest.bat to include the classpath to GSON

heejet committed:
Refactor code to create a Storage class to handle the loading of
data.

heejet committed:
Fix bug where storage still tries to load file despite it being
empty.

heejet committed:
Add testing mode which clears all data saved for the script to test
from the start.

heejet committed:
Update documentation

heejet committed:
Refactor code

heejet committed:
Refactor code for A-MoreOOP

Currently all code to execute command in LUKE are stored as methods
Luke.java.

This method of coding does not fully utilize the capabilities of
java. Use of functional programming here will result in extremely
long files.

Extract a command class for all commands used in LUKE.
Rename TaskOrganizer to TaskList.
Rename Response to Ui.
Add Parser class.

By extracting each command into its own class, we can take use
polymorphism to handle the different execution of each command. It
also makes the code shorter and easier to read.

heejet committed:
Move save taskList method to Storage class

heejet committed:
Change command format to include delimiter "|" and
add Level-9: Find

heejet committed:
Bug fixes

heejet committed:
Merge pull request #1 from heejet/branch-Level-9

Branch level 9

heejet committed:
Fix bugs in testing mode

heejet committed:
Add documentation for command package

heejet committed:
Add documentation to exception package

heejet committed:
Add documentation to icon package

heejet committed:
Add documentation to task package

heejet committed:
Add documentation to luke

heejet committed:
Merge pull request #2 from heejet/branch-A-JavaDoc

Branch a java doc

mathanmahe commented on src/main/java/luke/Luke.java:
```diff
@@ -0,0 +1,69 @@
+ * <code>Parser</code> object to execute various commands from the user.
+ */
+public class Luke {
+    /** An object to handle the user interaction with Luke */
```
Good job writing descriptive header comments, and following proper layout for comments relative to their position in code

mathanmahe commented on src/main/java/luke/Parser.java:
```diff
@@ -0,0 +1,216 @@
+     * @return A Command object that executes the command corresponding to the command keyword keyed in by the user.
+     */
+    private Command prepareCommand(String command, String description, TaskList tasks) {
+        switch (command) {
```
Good job on using abstraction to improve readability.

mathanmahe commented on src/main/java/luke/TaskList.java:
```diff
@@ -0,0 +1,229 @@
+    }
+
+    public ArrayList<Task> findTask(String toFind) {
+        ArrayList<Task> found = new ArrayList<>();
```
Code Standard: Use plural forms for collection of objects.

Can consider renaming to `foundTasks`.

mathanmahe commented on src/main/java/luke/TaskList.java:
```diff
@@ -0,0 +1,229 @@
+     * @return ArrayList of Tasks.
+     */
+    public ArrayList<Task> getTaskList() {
+        ArrayList<Task> copy = new ArrayList<Task>();
```
Jave coding standard: Use plural form for collection of objects.

As mentioned above, can give a suitable name in plural form.

mathanmahe commented on src/main/java/luke/command/Command.java:
```diff
@@ -0,0 +1,22 @@
+package luke.command;
```
Good application of OOP!

mathanmahe reviewed (commented):
Well done! Very well written code and use of OOP, with good adherence to Java coding standards and code quality. I have left a few minor suggestions!

heejet committed:
Refactor code

heejet committed:
Add A-UserGuide

heejet committed:
Add Level-8: Dates and Times

heejet committed:
Merge pull request #3 from heejet/branch-Level-8

Add Level-8: Dates and Times

heejet committed:
Fix bug where serial number for list resets from 1 whenever find
command is used

heejet committed:
Update user guide for Level-8

heejet committed:
Update user guide
